src/fio/fioEngine.js

```diff
@@ -287,192 +264,220 @@ export class FioEngine extends CurrencyEngine {
+          notes
         }
-        this.addTransaction(currencyCode, edgeTransaction)
       }
+      this.addTransaction(currencyCode, edgeTransaction)
     }
 
     return action.block_num
   }
 
-  async checkTransactions(): Promise<boolean> {
+  async checkTransactions(historyNodeIndex: number = 0): Promise<boolean> {
     let newHighestTxHeight = this.walletLocalData.otherData.highestTxHeight
     let lastActionSeqNumber = 0
     const fioSDK = new FIOSDK(
       '',
       '',
-      this.currencyInfo.defaultSettings.historyNodeUrls[0],
+      this.currencyInfo.defaultSettings.historyNodeUrls[historyNodeIndex],
       this.fetchCors,
       undefined,
       this.tpid
     )
     const actor = fioSDK.transactions.getActor(this.walletInfo.keys.publicKey)
     try {
-      const lastActionObject = await this.multicastServers(
-        'history',
+      const lastActionObject = await this.requestHistory(
+        historyNodeIndex,
         {
           account_name: actor,
           pos: -1,
           offset: -1
         },
-        this.currencyInfo.defaultSettings.historyNodeActions.getActions
+        HISTORY_NODE_ACTIONS.getActions
       )
 
-      if (lastActionObject.actions && lastActionObject.actions.length) {
+      asHistoryResponse(lastActionObject)
+      if (lastActionObject.actions.length) {
+        asHistoryNodeAction(lastActionObject.actions[0])
         lastActionSeqNumber = lastActionObject.actions[0].account_action_seq
+      } else {
+        // if no transactions at all
+        return true
+      }
+      if (lastActionObject.error && lastActionObject.error.noNodeForIndex) {
+        return false
       }
     } catch (e) {
-      console.log(e)
-      this.log(e)
-    }
-
-    if (!lastActionSeqNumber) {
-      return true
+      return this.checkTransactions(++historyNodeIndex)
     }
 
-    const limit = 10
-    let offset = 0
-
-    // history node saves up to 1000 records, so if seq number is more than 1000 we set pos to smallest value
-    if (lastActionSeqNumber > 1000) {
-      offset = lastActionSeqNumber - 1000
-    }
-
-    while (true) {
-      this.log('looping through checkTransactions')
-      const actionsObject = await this.multicastServers(
-        'history',
-        {
-          account_name: actor,
-          pos: offset,
-          offset: limit
-        },
-        this.currencyInfo.defaultSettings.historyNodeActions.getActions
-      )
-      let actions = []
-      actionsObject.actions.sort((a, b) => a.block_num - b.block_num)
+    const offset = 20
+    let pos = lastActionSeqNumber
+    let finish = false
 
-      if (actionsObject.actions && actionsObject.actions.length > 0) {
-        actions = actionsObject.actions
-      } else {
+    while (!finish) {
+      if (pos < 0) {
         break
       }
+      let actionsObject
+      try {
+        actionsObject = await this.requestHistory(
+          historyNodeIndex,
+          {
+            account_name: actor,
+            pos,
```

Comment:
confirm if history node throw an error (or what response it would be) if seq number is 2000 for example and loop is went to pos = 1000 and I'm trying to get actions with such params: pos = 1000, offset = -20